src/hotspot/share/services/threadService.cpp

```diff
@@ -1473,6 +1473,9 @@ oop ThreadSnapshotFactory::get_thread_snapshot(jobject jthread, TRAPS) {
     }
   } else {
     java_thread = java_lang_Thread::thread(thread_h());
+    if (java_thread == nullptr) {
+        return nullptr; // thread terminated
```

Comment:
If you return here what does that mean for the null check at line 1483? Is that code now dead?

Reply:
Here, we have this extra null check when is_virtual is false.  If not is_virtual, we really need a java_thread, or give up.
Down at 1483 we might have have is_virtual true, and we also may have found a java_thread, so I think we need both checks.